spring-cloud-gcp-samples/spring-cloud-gcp-trace-sample/src/test/java/com/example/TraceSampleApplicationIntegrationTests.java

```diff
@@ -82,17 +95,42 @@
 
   private TraceServiceBlockingStub traceServiceStub;
 
-  @BeforeClass
-  public static void checkToRun() {
-    assumeThat(
-        "Google Cloud Trace integration tests are disabled. "
-            + "Please use '-Dit.trace=true' to enable them. ",
-        System.getProperty("it.trace"),
-        is("true"));
+  static final String projectName =
+      ProjectName.of(ServiceOptions.getDefaultProjectId()).getProject();
+
+  private static TopicAdminClient topicAdminClient;
+
+  private  static SubscriptionAdminClient subscriptionAdminClient;
+
+  private static PubSubAdmin pubSubAdmin;
+
+  @BeforeAll
+  static void setup() throws IOException {
+
+    projectIdProvider = () -> projectName;
+
+    topicAdminClient = TopicAdminClient.create();
+    subscriptionAdminClient = SubscriptionAdminClient.create();
```

Comment:
Similarly, since `topicAdminClient` and `subscriptionAdminClient` are only used here, they can become local variables. The smaller the variable scope, the less context we have to keep in mind when reading code.